File: 0463-island-perimeter/0463-island-perimeter.py
class Solution:
  def islandPerimeter(self, grid: List[List[int]]) -> int:
    maxy = len(grid) - 1
    maxx = len(grid[0]) - 1
    res = 0
    for y in range(maxy + 1):
      for x in range(maxx + 1):
        if grid[y][x]:
          if x == 0 or not grid[y][x - 1]:
            res += 1
          if x == maxx or not grid[y][x + 1]:
            res += 1
          if y == 0 or not grid[y - 1][x]:
            res += 1
          if y == maxy or not grid[y + 1][x]:
            res += 1
    return res
        

File: 0463-island-perimeter/0463-island-perimeter.ts
function islandPerimeter(grid: number[][]): number {
  let ans = 0;
  for (let y = 0; y < grid.length; y++) {
    for (let x = 0; x < grid[0].length; x++) {
      if (!grid[y][x]) continue;
      if (!y || !grid[y-1][x]) ans++;
      if (!x || !grid[y][x-1]) ans++;
      if (y === grid.length - 1 || !grid[y + 1][x]) ans++;
      if (x === grid[0].length - 1 || !grid[y][x + 1]) ans++;
    }
  }
  return ans;
};